ports/rp2/libm_extra.c

```diff
@@ -0,0 +1,59 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2023 Damien P. George
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include <math.h>
+#include <stdint.h>
+
+#define INF_POS (0x7f800000)
+#define INF_NEG (0xff800000)
+
+union float_int_t {
+    float f;
+    uint32_t i;
+};
+
+float __wrap___aeabi_fadd(float x, float y);
+
+float __aeabi_fadd(float x, float y) {
+    union float_int_t xu = {.f = x};
+    union float_int_t yu = {.f = y};
+    if ((xu.i << 1) == (INF_POS << 1)) {
```

Comment:
This shift is clever but it results in quite a few instructions, see https://gcc.godbolt.org/z/vPKhq5f7M 

Suggest falling back on the property that (INFINITY + anything_but_INFINITY == INFINITY) and doing something like this: https://gcc.godbolt.org/z/Tsv9783aq

Reply:
I tried a few variants and I like my version better :)  it has the following properties:
- stack usage is smaller, only two registers are pushed
- in the very common case that x is a finite number, there's only one comparison before it defers to `__wrap___aeabi_fadd`

Reply:
I added comments to describe this.

Also:

> INFINITY + anything_but_INFINITY == INFINITY

That's not true: INF + NAN == NAN. So it'd require more logic to handle that.